Agoda.Frameworks.LoadBalancing/ResourceManager.cs

```diff
@@ -22,12 +22,19 @@ public interface IResourceManager<TSource>
 
     public static class ResourceManager
     {
-        public static IResourceManager<TSource> Create<TSource>(IEnumerable<TSource> sources) =>
-            new ResourceManager<TSource>(
-                sources.ToDictionary(
-                    x => x,
-                    _ => WeightItem.CreateDefaultItem()),
-                new AgodaWeightManipulationStrategy());
+        public static IResourceManager<TSource> Create<TSource>(IEnumerable<TSource> sources)
+        {
+            var resources = new Dictionary<TSource, WeightItem>();
```

Comment:
using ConcurrentDictionary, you can make this dictionary thread-safety as well as can use .GetOrAdd() syntax that will make this code shorter